test/integration/integration_suite_test.go

```diff
@@ -212,19 +212,30 @@ var _ = AfterSuite(func(done Done) {
 		close(done)
 		return
 	}
+
+	if kubeCtl == nil {
+		logf.Log.Info("No cluster found")
+		close(done)
+		return
+	}
+
 	var controllersCored *string
 	var err error
+	kubeCtl.SetContext(clusterContext)
 	cluster := clusterContext
-	kubeCtl.SetContext(cluster)
 	if len(cluster) == 0 {
 		cluster = "default"
 	}
 
 	By(cluster + ": Check for controllers' restarts")
 	err = utils.CheckRestart(kubeCtl)
 	if err != nil {
-		logf.Log.Error(err, "error restarting nephe controller")
-		controllersCored = &cluster
+		if apierrors.IsNotFound(err) {
+			logf.Log.Info("nephe controller not deployed")
+		} else {
+			logf.Log.Error(err, "error restarting nephe controller")
```

Comment:
if there is an error, should we return?

Reply:
Not really, this error indicates that the nephe controller has restarted during the test (pod restart count > 0). In this case we continue the AfterSuite clean up and collect support bundle or preserve setup later. At the end the test will error out.